tempesta_fw/http_msg.c

```diff
@@ -948,6 +948,9 @@ tfw_http_msg_free(TfwHttpMsg *m)
 	if (!m)
 		return;
 
+	/* Check that the paired response was destroyed before request. */
+	if (m->conn && (TFW_CONN_TYPE(m->conn) & Conn_Clnt))
+		WARN_ON_ONCE(m->pair);
```

Comment:
Usually we don't write assertions this way, but place the whole condition into the assertion.